stores/directory/local.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod, abstractstaticmethod
+from pathlib import Path
+from typing import List
+
+from stores.directory.base import BaseWritableSingleDirectoryStore
+
+
+class LocalDirectoryStore(BaseWritableSingleDirectoryStore):
+
+    local_path: Path
+
+    def __init__(self, local_dir):
+        # Takes a path or a string representing a path as input
+
+        # If it is not a path, pathify it
+        if type(local_dir) != Path:
```

Comment:
works but use `if not isinstance(local_dir, Path)`

in fact if you make it

```
if not isinstance(local_dir, Path):
      local_dir_path = Path(local_dir)
 else:
      local_dir_path = local_dir
```

That'll make vscode automcompetion (and pydantic if we ever go there) happy as it'll have a concrete type.

I wouldn't bother with the try catch, Pathlib should be able to tell us if local_dir is an invalid input.